contract_interface.md

```diff
@@ -24,6 +25,8 @@ On successful execution, the code should return via a normal code path.
 
 If it needs to abort due to a failure, an *unreachable* instruction should be executed.
 
+Further more, the ewasm spec disallows the use of a 'start' function in an ewasm contract, because it is executed upon module instantiation.
```

Comment:
In pwasm we decided to leave it as is. The reason for that we don't want to introduce unnecessary complication s since `start` function might still be used by the wasm producer. On the other hand, the logic is pretty much the same, consider this pseudo-code:

```
// let's say `env` is a EEI analog

try {
  // this will implicitly run the `start` function
  instance = instantiate_module(module, env)
  instance.invoke(env, "entrypoint")
} catch {
  // ...
}
```

there is almost no difference between running or not running the start function.

Reply:
Our decision to remove the start function was based on some detail that came up in implementing ewasm-kernel that required something to be done between instantiation and execution. @axic can elaborate a bit more on this.

I think that it seems to be a bit of a workaround that should be streamlined, so it may be revisited in the future.

Reply:
> because it is executed upon module instantiation

This explanation is a little vague, and I can foresee it being the source of future confusion if left as-is. @axic could you perhaps flesh this out?

Reply:
@lrettig I was not involved in writing ewasm-kernel, hence the shallowness of the explanation :)